actionmailer/lib/action_mailer/railtie.rb

```diff
@@ -42,7 +42,9 @@ class Railtie < Rails::Railtie # :nodoc:
         register_observers(options.delete(:observers))
 
         if delivery_job = options.delete(:delivery_job)
-          self.delivery_job = delivery_job.constantize
+          ActiveSupport.on_load(:active_job) do
```

Comment:
```
delivery_job = options.delete(:delivery_job)
ActiveSupport.on_load(:active_job) do
  ::ActionMailer::Base.delivery_job ||= delivery_job ? delivery_job.constantize : ::ActionMailer::MailDeliveryJob
end
```

I think the `if` needs to go away to make sure we always set this value

Reply:
Nice catch! I removed the `if ...`